scheduler/scheduler_test.go

```diff
@@ -45,13 +45,17 @@ type mockMetricManager struct {
 	failValidatingMetricsAfter int
 	failuredSoFar              int
 	autodiscoverPaths          []string
+	timeToWait                 time.Duration
 }
 
 func (m *mockMetricManager) StreamMetrics(string, map[string]map[string]string, time.Duration, int64) (chan []core.Metric, chan error, []error) {
 	return nil, nil, nil
 }
 
 func (m *mockMetricManager) CollectMetrics(string, map[string]map[string]string) ([]core.Metric, []error) {
+	if m.timeToWait != 0 {
+		time.Sleep(m.timeToWait)
+	}
```

Comment:
This 'if' statement is not necessary - you can straightforward do `time.(m.timeToWait)